gen/api_defs.jl

```diff
@@ -155,13 +155,13 @@
 @bind h5p_get_fapl_mpio32(fapl_id::hid_t, comm::Ptr{Hmpih32}, info::Ptr{Hmpih32})::herr_t "Error getting MPIO properties"
 @bind h5p_get_fapl_mpio64(fapl_id::hid_t, comm::Ptr{Hmpih64}, info::Ptr{Hmpih64})::herr_t "Error getting MPIO properties"
 @bind h5p_get_fclose_degree(fapl_id::hid_t, fc_degree::Ref{Cint})::herr_t "Error getting close degree"
-@bind h5p_get_filter_by_id(plist_id::hid_t, filter_id::H5Z_filter_t, flags::Ref{Cuint}, cd_nelmts::Ref{Csize_t}, cd_values::Ptr{Cuint}, namelen::Csize_t, name::Ptr{UInt8}, filter_config::Ptr{Cuint})::herr_t "Error getting filter ID"
+@bind h5p_get_filter_by_id(plist_id::hid_t, filter_id::Cint, flags::Ref{Cuint}, cd_nelmts::Ref{Csize_t}, cd_values::Ptr{Cuint}, namelen::Csize_t, name::Ptr{UInt8}, filter_config::Ptr{Cuint})::herr_t "Error getting filter ID"
```

Comment:
Why delete the definition for `H5Z_filter_t` ?

Reply:
It struck me as out of place that this is one of the only places where a const name was given to a C typedef rather than just using the underlying C type directly (excluding the much-used `herr_t`, `hid_t`, etc.). I have no issue putting it back, especially if you think we should move in the direction of adding more type aliases (such as named constants for all of the enum types and/or just using `@enum` or `@cenum`).